theano/gpuarray/dnn.py

```diff
@@ -298,18 +298,18 @@ def __init__(self, border_mode, conv_mode='conv',
     def make_node(self, kern_shape, subsample=(1, 1, 0)):
         if kern_shape.type.ndim != 1 or kern_shape.type.dtype != 'int64':
             raise TypeError('kern must be 1D shape tensor')
-        if isinstance(border_mode, integer_types):
-            border_mode = (border_mode,) * len(subsample)
-        if isinstance(border_mode, tuple):
-            assert len(border_mode) == len(subsample)
-            border_mode = tuple(map(int, border_mode))
-        if not ((isinstance(border_mode, tuple) and min(border_mode) >= 0) or
-                border_mode in ('valid', 'full', 'half')):
+        if isinstance(self.border_mode, integer_types):
```

Comment:
The real fix is to remove the border_mode from the \__init method and put it in the make_node() method.